File: fabric-frontend/src/app/api/models/console-response-item-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ConsoleResponseDescriptionDto } from './console-response-description-dto';
import { NavItem } from './nav-item';
import { TabularResponse } from './tabular-response';
export interface ConsoleResponseItemDto {
  consoleOutputType?: 'JSON' | 'MESSAGE' | 'STACKTRACE' | 'TABULAR';
  content?: string;
  description?: ConsoleResponseDescriptionDto;
  navItems?: Array<NavItem>;
  systemType?: string;
  tabular?: TabularResponse;
}

File: fabric-backend-core/src/main/java/io/storydoc/fabric/console/app/query/ConsoleResponseItemDTO.java
package io.storydoc.fabric.console.app.query;

import io.storydoc.fabric.console.app.describe.ConsoleOutputType;
import io.storydoc.fabric.console.app.navigation.NavItem;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ConsoleResponseItemDTO {
    private String systemType;
    private ConsoleOutputType consoleOutputType;
    private ConsoleResponseDescriptionDTO description;
    private String content;
    private TabularResponse tabular;
    private List<NavItem> navItems;
}
